vm/src/stdlib/itertools.rs

```diff
@@ -1952,4 +1955,79 @@ mod decl {
             Ok(PyIterReturn::Return(vm.new_tuple((old, new)).into()))
         }
     }
+
+    #[pyattr]
+    #[pyclass(name = "batched")]
+    #[derive(Debug, PyPayload)]
+    struct PyItertoolsBatched {
+        exhausted: AtomicCell<bool>,
+        iterable: PyIter,
+        n: AtomicCell<usize>,
+    }
+
+    #[derive(FromArgs)]
+    struct BatchedNewArgs {
+        #[pyarg(positional)]
+        iterable_ref: PyObjectRef,
+        #[pyarg(positional)]
+        n: PyIntRef,
+    }
+
+    impl Constructor for PyItertoolsBatched {
+        type Args = BatchedNewArgs;
+
+        fn py_new(
+            cls: PyTypeRef,
+            Self::Args { iterable_ref, n }: Self::Args,
+            vm: &VirtualMachine,
+        ) -> PyResult {
+            let n = n.as_bigint();
+            if n.lt(&BigInt::one()) {
+                return Err(vm.new_value_error("n must be at least one".to_owned()));
+            }
+            let n = n.to_usize().unwrap();
```

Comment:
What happens if n is a negative number? That will be a python error rather than panic (by unwrap).
By the error type and message, you might want to change the type of `BatchedNewArgs::n` to `usize` or any unsigned type. Otherwise making an `ok_or_else` chain will be a good choice.

Reply:
Doesn't the conditional on line 1985 ensure that n will not be a number less than 1?

Reply:
That's right, thank you! How about n > usize::MAX?
Since unwrap is one of main source of uncontrolled panic, we prefer to unsure they don't have edge cases and a bit picky about it.
If that's logically not able to be triggered, using `expect()` with reasoning  rather than `unwrap()` will be very helpful to provide the reasoning and a debug hint for future regression.

Reply:
Great point! CPython throws the following error in such case: `OverflowError: Python int too large to convert to C ssize_t`.

I've modified the unwrap to instead use ok_or and return an overflow error in this case.